api/v1alpha1/nodepool_types.go

```diff
@@ -419,11 +420,81 @@ type PowerVSNodePoolPlatform struct {
 	ImageDeletePolicy string `json:"imageDeletePolicy,omitempty"`
 }
 
+// KubevirtCompute contains values associated with the virtual compute hardware requested for the VM.
+type KubevirtCompute struct {
+	// Memory represents how much guest memory the VM should have
+	//
+	// +kubebuilder:default="4Gi"
+	Memory *resource.Quantity `json:"memory"`
+
+	// Cores represents how many cores the guest VM should have
+	//
+	// +kubebuilder:default=2
+	Cores *uint32 `json:"cores"`
+}
+
+// KubevirtPersistentVolume containes the values involved with provisioning persistent storage for a KubeVirt VM.
+type KubevirtPersistentVolume struct {
+	// Size is the size of the persistent storage volume
+	//
+	// +kubebuilder:default="16Gi"
+	Size *resource.Quantity `json:"size"`
+	// StorageClass is the storageClass used for the underlying PVC that hosts the volume
+	//
+	// +optional
+	StorageClass *string `json:"storageClass,omitempty"`
+}
+
+// KubevirtRootVolume represents the volume that the rhcos disk will be stored and run from.
+type KubevirtRootVolume struct {
+	// Image represents what rhcos image to use for the node pool
+	Image *KubevirtDiskImage `json:"diskImage"`
+
+	// KubevirtVolume represents of type of storage to run the image on
+	KubevirtVolume `json:",inline"`
+}
+
+// KubevirtVolumeType is a specific supported KubeVirt volumes
+//
+// +kubebuilder:validation:Enum=Persistent
+type KubevirtVolumeType string
+
+const (
+	// KubevirtVolumeTypePersistent represents persistent volume for kubevirt VMs
+	KubevirtVolumeTypePersistent KubevirtVolumeType = "Persistent"
+)
+
+// KubevirtVolume represents what kind of storage to use for a KubeVirt VM volume
+type KubevirtVolume struct {
+	// Type represents the type of storage to associate with the kubevirt VMs.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:default=Persistent
+	Type KubevirtVolumeType `json:"type"`
+
+	// Persistent volume type means the VM's storage is backed by a PVC
+	// VMs that use persistent volumes can survive disruption events like restart and eviction
+	// This is the default type used when no storage type is defined.
+	//
+	// +optional
+	Persistent *KubevirtPersistentVolume `json:"persistent,omitempty"`
+}
+
+// KubevirtDiskImage contains values representing where the rhcos image is located
+type KubevirtDiskImage struct {
+	// ContainerDiskImage is a string representing the container image that holds the root disk
+	ContainerDiskImage *string `json:"containerDiskImage"`
```

Comment:
unless you need to differentiate between nil and "" I don't think this should be a pointer.

Reply:
I'd rather keep this as a pointer. I'm considering making this a union in the future where someone could specify a disk from an HTTP endpoint, or in a PVC, which would all get treated differently